src/main/java/at/ac/tuwien/kr/alpha/grounder/NaiveGrounder.java

```diff
@@ -290,24 +292,26 @@ public float computeAverageNoGoodLength() {
 				modifiedWorkingMemories.add(positiveStorage);
 			}
 		}
+
 		for (NonGroundRule nonGroundRule : rulesFromProgram) {
 			// Generate nogoods for all rules that are ground already.
 			if (!nonGroundRule.isOriginallyGround()) {
 				continue;
 			}
+
 			// Check if rule contains intervals (but is otherwise ground).
-			if (nonGroundRule.containsIntervals()) {
+			if (nonGroundRule.containsIntervals() || nonGroundRule.containsExternals()) {
```

Comment:
this might be a misunderstanding on my side, however: Just as an interval expands to multiple bindings/substitutions (for example `a(1..3)` will expand to `a(X)` for `X = 1`, `X = 2`, `X = 3`), an external atom might also expand to multiple bindings/substitutions (for example `&coolNumbers[X](3)` might result in the same three possible bindings for `X`). At least that#s how I understand it. So therefore, we do the same for intervals and externals: We "unwrap" them and get multiple substitutions.

Reply:
Ah, now I see. This place here is treating a special case, namely if the rule contains no ordinary atom with a variable that starts the grounding/substitution-finding of the rule. So I guess the check should not just be for the presence of an output-variable in the atom, but for the non-presence of any ordinary variable in an ordinary atom.

Maybe the code currently does no harm when called for all rules that contain external atoms. But I am not fully sure.

Reply:
OK. We should test this. If you can construct a test case that shows incorrectness of this implementation from the top of your head, this would be very helpful. I will try to construct one otherwise.

Probably then I need to just refine/reformulate the check to what you suggested.

Reply:
OK, I just had another look, and I don't see how this might lead to problems ... Could you provide a "counterexample", e.g. showcase how this behaves wrongly?

Reply:
I think it will work with the current implementation of the grounding, but might fail if we change that later. In any case it is unnecessary to do for all rules. A `// FIXME: should only be done for rules with externals where the rule contains no variable bound by an ordinary atom.` should be enough for now.